### ISSUE TITLE
Simple approach with promoterome produces viewpoints without TSS-digest

### ISSUE THREAD

**Issue by hansenp** [TheJacksonLaboratory/Gopher#350]
If I run the simple approach with hg19, DpnII and default parameters on all protein coding genes, GOPHER reports about 40 viewpoints to have no digest that contains the TSS.

```
[ERROR] 09-13-2018 14:26:47 [ForkJoinPool.commonPool-worker-7] (ViewPoint.java:607) - TSTD1 At least one digest must contain 'genomicPos' (chr1:161008774)
[ERROR] 09-13-2018 14:26:47 [ForkJoinPool.commonPool-worker-1] (ViewPoint.java:607) - SEMA4A At least one digest must contain 'genomicPos' (chr1:156123322)
[ERROR] 09-13-2018 14:26:47 [ForkJoinPool.commonPool-worker-7] (ViewPoint.java:607) - LRRC52 At least one digest must contain 'genomicPos' (chr1:165513478)
[ERROR] 09-13-2018 14:26:47 [Thread-4] (ViewPoint.java:607) - CCDC30 At least one digest must contain 'genomicPos' (chr1:42921741)
[ERROR] 09-13-2018 14:26:48 [ForkJoinPool.commonPool-worker-2] (ViewPoint.java:607) - NTNG1 At least one digest must contain 'genomicPos' (chr1:107682540)
[ERROR] 09-13-2018 14:26:48 [ForkJoinPool.commonPool-worker-4] (ViewPoint.java:607) - CYB561D1 At least one digest must contain 'genomicPos' (chr1:110036701)
[ERROR] 09-13-2018 14:26:53 [ForkJoinPool.commonPool-worker-4] (ViewPoint.java:607) - UCN3 At least one digest must contain 'genomicPos' (chr10:5406972)
[ERROR] 09-13-2018 14:26:54 [Thread-4] (ViewPoint.java:607) - PNLIPRP1 At least one digest must contain 'genomicPos' (chr10:118350471)
[ERROR] 09-13-2018 14:27:00 [ForkJoinPool.commonPool-worker-6] (ViewPoint.java:607) - PAAF1 At least one digest must contain 'genomicPos' (chr11:73587744)
[ERROR] 09-13-2018 14:27:00 [ForkJoinPool.commonPool-worker-4] (ViewPoint.java:607) - PAX6 At least one digest must contain 'genomicPos' (chr11:31825785)
[ERROR] 09-13-2018 14:27:00 [ForkJoinPool.commonPool-worker-6] (ViewPoint.java:607) - KRTAP5-10 At least one digest must contain 'genomicPos' (chr11:71276609)
[ERROR] 09-13-2018 14:27:00 [ForkJoinPool.commonPool-worker-1] (ViewPoint.java:607) - NLRP10 At least one digest must contain 'genomicPos' (chr11:7985059)
[ERROR] 09-13-2018 14:27:01 [ForkJoinPool.commonPool-worker-7] (ViewPoint.java:607) - LBHD1 At least one digest must contain 'genomicPos' (chr11:62439241)
[ERROR] 09-13-2018 14:27:06 [Thread-4] (ViewPoint.java:607) - ADCY6 At least one digest must contain 'genomicPos' (chr12:49182879)
[ERROR] 09-13-2018 14:27:17 [ForkJoinPool.commonPool-worker-0] (ViewPoint.java:607) - GOLGA6L9 At least one digest must contain 'genomicPos' (chr15:82804769)
[ERROR] 09-13-2018 14:27:18 [ForkJoinPool.commonPool-worker-0] (ViewPoint.java:607) - GOLGA6L9 At least one digest must contain 'genomicPos' (chr15:82722988)
[ERROR] 09-13-2018 14:27:27 [ForkJoinPool.commonPool-worker-5] (ViewPoint.java:607) - CAMKK1 At least one digest must contain 'genomicPos' (chr17:3796337)
[ERROR] 09-13-2018 14:27:28 [ForkJoinPool.commonPool-worker-6] (ViewPoint.java:607) - ACBD4 At least one digest must contain 'genomicPos' (chr17:43213014)
[ERROR] 09-13-2018 14:27:34 [ForkJoinPool.commonPool-worker-7] (ViewPoint.java:607) - ZNF578 At least one digest must contain 'genomicPos' (chr19:52956829)
[ERROR] 09-13-2018 14:27:43 [ForkJoinPool.commonPool-worker-5] (ViewPoint.java:607) - SMARCAL1 At least one digest must contain 'genomicPos' (chr2:217277473)
[ERROR] 09-13-2018 14:27:43 [ForkJoinPool.commonPool-worker-2] (ViewPoint.java:607) - SCRN3 At least one digest must contain 'genomicPos' (chr2:175260457)
[ERROR] 09-13-2018 14:27:47 [ForkJoinPool.commonPool-worker-5] (ViewPoint.java:607) - PIGP At least one digest must contain 'genomicPos' (chr21:38445458)
[ERROR] 09-13-2018 14:27:49 [Thread-4] (ViewPoint.java:607) - GAS2L1 At least one digest must contain 'genomicPos' (chr22:29702985)
[ERROR] 09-13-2018 14:27:49 [ForkJoinPool.commonPool-worker-7] (ViewPoint.java:607) - GSTT2B At least one digest must contain 'genomicPos' (chr22:24322314)
[ERROR] 09-13-2018 14:27:49 [ForkJoinPool.commonPool-worker-3] (ViewPoint.java:607) - TMEM191B At least one digest must contain 'genomicPos' (chr22:20377669)
[ERROR] 09-13-2018 14:27:57 [Thread-4] (ViewPoint.java:607) - GPR15 At least one digest must contain 'genomicPos' (chr3:98250743)
[ERROR] 09-13-2018 14:27:57 [ForkJoinPool.commonPool-worker-5] (ViewPoint.java:607) - ZBTB38 At least one digest must contain 'genomicPos' (chr3:141105288)
[ERROR] 09-13-2018 14:28:06 [ForkJoinPool.commonPool-worker-5] (ViewPoint.java:607) - BOD1L1 At least one digest must contain 'genomicPos' (chr4:13629328)
[ERROR] 09-13-2018 14:28:13 [ForkJoinPool.commonPool-worker-1] (ViewPoint.java:607) - ERAP1 At least one digest must contain 'genomicPos' (chr5:96149848)
[ERROR] 09-13-2018 14:28:13 [ForkJoinPool.commonPool-worker-0] (ViewPoint.java:607) - GZMK At least one digest must contain 'genomicPos' (chr5:54320107)
[ERROR] 09-13-2018 14:28:13 [ForkJoinPool.commonPool-worker-4] (ViewPoint.java:607) - C5orf67 At least one digest must contain 'genomicPos' (chr5:55902059)
[ERROR] 09-13-2018 14:28:21 [ForkJoinPool.commonPool-worker-3] (ViewPoint.java:607) - RBM24 At least one digest must contain 'genomicPos' (chr6:17282932)
[ERROR] 09-13-2018 14:28:21 [ForkJoinPool.commonPool-worker-4] (ViewPoint.java:607) - MAP3K4 At least one digest must contain 'genomicPos' (chr6:161412759)
[ERROR] 09-13-2018 14:28:21 [ForkJoinPool.commonPool-worker-7] (ViewPoint.java:607) - EXOC2 At least one digest must contain 'genomicPos' (chr6:693141)
[ERROR] 09-13-2018 14:28:28 [ForkJoinPool.commonPool-worker-7] (ViewPoint.java:607) - CREB5 At least one digest must contain 'genomicPos' (chr7:28338940)
[ERROR] 09-13-2018 14:28:28 [ForkJoinPool.commonPool-worker-2] (ViewPoint.java:607) - AMZ1 At least one digest must contain 'genomicPos' (chr7:2727836)
[ERROR] 09-13-2018 14:28:28 [ForkJoinPool.commonPool-worker-7] (ViewPoint.java:607) - STEAP2 At least one digest must contain 'genomicPos' (chr7:89841174)
[ERROR] 09-13-2018 14:28:33 [ForkJoinPool.commonPool-worker-3] (ViewPoint.java:607) - RAD54B At least one digest must contain 'genomicPos' (chr8:95449180)
[ERROR] 09-13-2018 14:28:33 [ForkJoinPool.commonPool-worker-3] (ViewPoint.java:607) - FSBP At least one digest must contain 'genomicPos' (chr8:95449180)
[ERROR] 09-13-2018 14:28:38 [ForkJoinPool.commonPool-worker-4] (ViewPoint.java:607) - RAD23B At least one digest must contain 'genomicPos' (chr9:110047050)
[ERROR] 09-13-2018 14:28:38 [ForkJoinPool.commonPool-worker-4] (ViewPoint.java:607) - SLC31A2 At least one digest must contain 'genomicPos' (chr9:115913238)
[ERROR] 09-13-2018 14:28:45 [ForkJoinPool.commonPool-worker-4] (ViewPoint.java:607) - SUV39H1 At least one digest must contain 'genomicPos' (chrX:48553945)
[ERROR] 09-13-2018 14:28:45 [ForkJoinPool.commonPool-worker-5] (ViewPoint.java:607) - ASMT At least one digest must contain 'genomicPos' (chrX:1733941)
[ERROR] 09-13-2018 14:28:45 [ForkJoinPool.commonPool-worker-7] (ViewPoint.java:607) - MAGEA11 At least one digest must contain 'genomicPos' (chrX:148793738)
[ERROR] 09-13-2018 14:28:45 [Thread-4] (ViewPoint.java:607) - GNL3L At least one digest must contain 'genomicPos' (chrX:54556644)
[ERROR] 09-13-2018 14:28:47 [ForkJoinPool.commonPool-worker-2] (ViewPoint.java:607) - ASMT At least one digest must contain 'genomicPos' (chrY:1683941)
```

**Comment by pnrobinson** [comment]
Isn't this expected behaviour? Especially with DpnII there will be TSS with no valid frag?

**Comment by hansenp** [comment]
No, I don't think so. All viewpoints I looked at have one thing in common. The TSS positions equals a start or end position of a segment. One example:

![image](https://user-images.githubusercontent.com/10495485/45503354-e7f69e00-b786-11e8-86a6-c263a9ac731b.png)

The starting position of the fifth segment and the position of the TSS are the same.

**Comment by hansenp** [comment]
I changed the line

`filter(segment -> segment.getStartPos() < genomicPos && segment.getEndPos() >= genomicPos).`

in class `ViewPoint` to

`filter(segment -> segment.getStartPos() <= genomicPos && segment.getEndPos() >= genomicPos).`

Otherwise this condition becomes false, if `genomicPos` overlaps the first position of a segment. I tested this for the promoterome. No errors occurred. Pushed this to develop.
